pulp_rpm/app/tasks/signing.py

```diff
@@ -66,3 +78,62 @@ def sign_and_create(
     if "upload" in data:
         del data["upload"]
     general_create(app_label, serializer_name, data=data, context=context, *args, **kwargs)
+
+
+def signed_add_and_remove(
+    repository_pk, add_content_units, remove_content_units, base_version_pk=None
+):
+    repo = RpmRepository.objects.get(pk=repository_pk)
+
+    if repo.package_signing_service:
+        # sign each package and replace it in the add_content_units list
+        signed_packages = []
+
+        for package in Package.objects.filter(pk__in=add_content_units):
+            content_artifact = package.contentartifact_set.first()
+            artifact_obj = content_artifact.artifact
```

Comment:
Is there an assumption here that the package is not `on_demand` or is that being checked somewhere else?